src/pages/Overall.js

```diff
@@ -0,0 +1,145 @@
+            isDesktop: false
+        }
+        this.updatePredicate = this.updatePredicate.bind(this)
+
+    }
+
+    componentDidMount() {
+        this.updatePredicate();
+        window.addEventListener("resize", this.updatePredicate);
+    }
+
+    updatePredicate() {
+        this.setState({ isDesktop: window.innerWidth > 750 });
+    }
+
+    changeHandler = (e) => {
+        const filter = e.target.value
+        this.setState({ filterType: filter })
+    }
+
+    searchHandler = (e) => {
+        var searchQuery = e.target.value
+
+        if (this.state.filterType === "region") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.region.toLowerCase().indexOf(searchQuery.toLowerCase()) !== -1
+            )
+            this.setState({ searchedRegions: searchResult })
+        } else if (this.state.filterType === "lastinput") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.lastinput.toString().indexOf(searchQuery.toString()) !== -1
+            )
+            this.setState({ searchedRegions: searchResult })
+        }
+        else if (this.state.filterType === "formnumbers") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.formnumbers.toString().indexOf(searchQuery) !== -1
+            )
+            this.setState({ searchedRegions: searchResult })
+        } else if (this.state.filterType === "voternumbers") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.voternumbers.toString().indexOf(searchQuery) !== -1
+            )
+            this.setState({ searchedRegions: searchResult })
+        } else if (this.state.filterType === "update") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.update.toString().indexOf(searchQuery) !== -1
+            )
+            this.setState({ searchedRegions: searchResult })
+        }
+    }
+    render() {
+        const isDesktop = this.state.isDesktop;
+        return (
+            <>
+                {isDesktop ? (<div className="Data">
+                    <div className="Overall-White-Container">
+                        <div className="filter">
+                            <div className="input-class">
+                                <select class="custom-select" style={{ width: "200px" }} required value={this.state.filterType} onChange={this.changeHandler}>
+                                    <option className="select-items" defaultValue>Filter</option>
+                                    <option value="region">Regions</option>
+                                    <option value="lastinput">Last Input</option>
+                                    <option value="formnumbers">Form Numbers</option>
+                                    <option value="voternumbers">Voter Numbers</option>
+                                    <option value="update">Update</option>
```

Comment:
why don't just define theses options as an array variable and call `map` on the array here.